xapian-letor/api/featurevector_internal.cc

```diff
@@ -117,3 +117,31 @@ FeatureVector::Internal::get_nonzero_num(){
     //cout << "nonzero: "<< nonzero <<endl;
     return nonzero;
 }
+
+void
+FeatureVector::Internal::normalise() {
+
+    int num_features = 19;
+    double temp = 0.0;
+    double max[num_features];
+
+    for(int i=0; i<19; ++i)
+    max[i] = 0.0;
+
+    for(int j=0; j<19; ++j) {
+        if(max[j] < fvals.find(j)->second)
+        max[j] = fvals.find(j)->second;
+    }
+
+     /* We have the maximum value of each feature overall.
+        Now we need to normalize each feature value of a featureVector
+        by dividing it by the corresponding max of the feature value
+    */
+    for(int j=0; j<19; ++j) {
+        temp = fvals.find(j)->second;
+        temp /= max[j];
+        fvals.insert(pair<int,double>(j,temp));
```

Comment:
Thanks for pointing this out. So stupid of me, there wasn't any normalisation taking place up until now.